[Proj4] Update to 7.1.1

Matioupi opened the pull request:
Bump proj4 version to 7.1.1

- What does your PR fix? Fixes #

- Which triplets are supported/not supported? Have you updated the CI baseline?
x64-windows tested locally

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

it should. I've seen PR #13428 on the same subject.

Matioupi committed:
fix for compile errors when building opencv with command line :
.\vcpkg.exe install opencv[nonfree,ade,contrib,cuda,dnn,eigen,ffmpeg,gdcm,ipp,jasper,jpeg,openexr,opengl,openmp,ovis,png,qt,sfm,tbb,tiff,vtk,webp,halide]:x64-windows
error was located in modules/imgcodecs/src/grfmt_jpeg2000.cpp file and is fixed by a new patch

Matioupi committed:
bumped port-version to 2 as required in https://github.com/microsoft/vcpkg/pull/14101#issuecomment-711944502

BillyONeal committed:
Merge remote-tracking branch 'origin/master' into HEAD

Matioupi committed:
bumped version to 7.1.1

Matioupi committed:
Merge remote-tracking branch 'upstream/master' into proj4_711

NancyLi1013 commented on ports/proj4/CONTROL:
```diff
@@ -1,9 +1,9 @@
-Version: 6.3.1
-Port-Version: 2
+Version: 7.1.1
+Port-Version: 3
```
```suggestion
```
This field is not needed, since this is a new version update.
Please see https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#update-the-port-version-field-in-the-manifest-file-of-any-modified-ports

Matioupi replied:
Done in https://github.com/microsoft/vcpkg/pull/14143/commits/8dc43c6370ae8db8c8be321c232573c7706756e9

NancyLi1013 commented on ports/proj4/portfile.cmake:
```diff
@@ -1,17 +1,12 @@
-        fix-sqlite-dependency-export.patch
-        fix-linux-build.patch
-        use-sqlite3-config.patch
+
```
```suggestion
```

Matioupi replied:
Done in https://github.com/microsoft/vcpkg/pull/14143/commits/8dc43c6370ae8db8c8be321c232573c7706756e9

Matioupi committed:
Remove port number as requested in https://github.com/microsoft/vcpkg/pull/14143#discussion_r509073854

Matioupi committed:
formatting change as requested in https://github.com/microsoft/vcpkg/pull/14143#discussion_r509074172

NancyLi1013 commented on ports/proj4/CONTROL:
```diff
@@ -1,9 +1,9 @@
-Version: 6.3.1
-Port-Version: 2
+Version: 7.1.1
+
```
```suggestion
```

NancyLi1013 committed:
Update ports/proj4/CONTROL

Matioupi committed:
Fixed CONTROL errors on linux

Matioupi committed:
Merge branch 'proj4_711' of https://github.com/Matioupi/vcpkg into proj4_711

Matioupi committed:
Fix more errors on x64-linux build tests

NancyLi1013 commented on ports/proj4/CONTROL:
```diff
@@ -1,11 +1,10 @@
 Homepage: https://github.com/OSGeo/PROJ
 Description: PROJ.4 library for cartographic projections
-Build-Depends: sqlite3[core]
+Build-Depends: sqlite3[core],tiff[tool],curl[core,tool]
```
```suggestion
Build-Depends: sqlite3[core], tiff[tool], curl[core,tool]
```

Matioupi replied:
Hi, I think I added the formatting changes, but latest commit also fixes x64-linux triplet build (mostly related to some build error on curl[tool] and tiff[tool] build errors on linux, that should probably also be addressed for those ports)

Matioupi committed:
Successfull build of x64-linux and x64-windows triplets

NancyLi1013 commented:
For `curl `on uwp
```
CMake Error at ports/curl/portfile.cmake:22 (message):
  schannel,sspi,tool are not supported on non-Windows and uwp platforms
Call Stack (most recent call first):
  scripts/ports.cmake:79 (include)
```
Since the feature `tool` is not supported on non-Windows and uwp platforms, the dependency should be updated as `curl[core, tool] (!uwp&windows)`

For `tiff `on uwp

```
D:\buildtrees\tiff\src\4.1.0-f65af59870.clean\tools\tiff2pdf.c(3453): error C4703: potentially uninitialized local pointer variable 'ac' used [D:\buildtrees\tiff\x64-uwp-dbg\tools\tiff2pdf.vcxproj]
    20>D:\buildtrees\tiff\src\4.1.0-f65af59870.clean\tools\tiff2pdf.c(3502): error C4703: potentially uninitialized local pointer variable 'lp' used [D:\buildtrees\tiff\x64-uwp-dbg\tools\tiff2pdf.vcxproj]
    20>D:\buildtrees\tiff\src\4.1.0-f65af59870.clean\tools\tiff2pdf.c(3504): error C4703: potentially uninitialized local pointer variable 'pt' used [D:\buildtrees\tiff\x64-uwp-dbg\tools\tiff2pdf.vcxproj]
```

Matioupi committed:
changed CONTROL as requested in https://github.com/microsoft/vcpkg/pull/14143#issuecomment-714266355

Matioupi commented on ports/proj4/CONTROL:
```diff
@@ -1,11 +1,10 @@
 Homepage: https://github.com/OSGeo/PROJ
 Description: PROJ.4 library for cartographic projections
-Build-Depends: sqlite3[core]
+Build-Depends: sqlite3[core], tiff[tool] (windows), tiff (!windows), curl[core, tool] (!uwp&windows)
```
From : https://proj.org/install.html#install-proj-dependencies both tiff and curl are dependencies of proj since version 7.
Still you are right that [tool]does not seems to be needed. I tested on a linux ubuntu 20.04 and the build success without adding tiff and curl, but that's probably because they are available system wide as system libs.

As for the other CI failures, the x64-uwp build is always failing with this kind of error :

`         gnom.cpp
         laea.cpp
         mod_ster.cpp
     6>C:\src\vcpkg\buildtrees\proj4\src\7.1.1-094056a32a.clean\src\filemanager.cpp(695,24): error C3861: 'CreateFileW': identifier not found [C:\src\vcpkg\buildtrees\proj4\x64-uwp-dbg\src\proj.vcxproj]
     6>C:\src\vcpkg\buildtrees\proj4\src\7.1.1-094056a32a.clean\src\filemanager.cpp(1215,34): error C2065: 'FOLDERID_LocalAppData': undeclared identifier [C:\src\vcpkg\buildtrees\proj4\x64-uwp-dbg\src\proj.vcxproj]
     6>C:\src\vcpkg\buildtrees\proj4\src\7.1.1-094056a32a.clean\src\filemanager.cpp(1215,13): error C3861: 'SHGetKnownFolderPath': identifier not found [C:\src\vcpkg\buildtrees\proj4\x64-uwp-dbg\src\proj.vcxproj]
     6>C:\src\vcpkg\buildtrees\proj4\src\7.1.1-094056a32a.clean\src\filemanager.cpp(1314,27): error C2065: 'GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS': undeclared identifier [C:\src\vcpkg\buildtrees\proj4\x64-uwp-dbg\src\proj.vcxproj]
     6>C:\src\vcpkg\buildtrees\proj4\src\7.1.1-094056a32a.clean\src\filemanager.cpp(1315,31): error C2065: 'GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT': undeclared identifier [C:\src\vcpkg\buildtrees\proj4\x64-uwp-dbg\src\proj.vcxproj]
     6>C:\src\vcpkg\buildtrees\proj4\src\7.1.1-094056a32a.clean\src\filemanager.cpp(1314,9): error C3861: 'GetModuleHandleEx': identifier not found [C:\src\vcpkg\buildtrees\proj4\x64-uwp-dbg\src\proj.vcxproj]
         nsper.cpp
         nzmg.cpp
         ortho.cpp`

in install-x64-uwp-dbg-out.log and I'm not getting a clue why this is happenning...

Matioupi replied:
oh, it seems that CreateFile does not exists for UWP

NancyLi1013 replied:
Yes, I noticed that you have posted this issue to upstream https://github.com/OSGeo/PROJ/issues/2391.
We might consider to update the changes in this PR if the issue is fixed on uwp.

Matioupi replied:
Hello, I gave a few more tries solving the compilation issues on uwp, but this requires more knowledge on the internal behaviour of proj4 and also the internal uwp compatible API's that I have and whish to acquire.
My opinion is that if the newest library version do not support uwp, I don't really see the point of having the vcpkg port supporting it. It's more a deprecation than a regression I believe.
If uwp compatibility is required in order to bump up the version, then it is probably faster to close this PR. At least I will not spend more time trying to patch for this. One solution would be for vcpkg to have a policy/way to enable multiple port versions with different plateform coverage, but if so, I missed it.

NancyLi1013 replied:
Sorry for the long delay. Our policy is to keep all triplets work normally if these triplets are supported on old version, except for the changes on the new version. For uwp as an example. if this is due to upstream change, which leads to the failure on uwp, we can disable the support for uwp.

NancyLi1013 commented:
Duplicated to #13428

Matioupi committed:
Added capability to build tools
Fixed internal PROJ_LIB database path which was currently set to package directory instead of install directory
Gave up support for uwp

Matioupi committed:
Added patch file missing from last commit

Matioupi committed:
More fixes for building tools on static mode

NancyLi1013 commented:
Hi @Matioupi 

Are you still interested in working done for this PR?

NancyLi1013 commented:
Thanks for the PR; we're closing this for now since there's been no response. If you'd like to continue working on it, please reopen and ping us!

Duplicated to #15978.
